meta-cli/src/cli/list.rs

```diff
@@ -0,0 +1,175 @@
+// Copyright Metatype OÜ, licensed under the Mozilla Public License Version 2.0.
+// SPDX-License-Identifier: MPL-2.0
+
+use super::serialize::SerializeReportExt;
+use super::{Action, ConfigArgs};
+use crate::config::{Config, PathOption};
+use crate::deploy::actors::console::ConsoleActor;
+use crate::deploy::actors::task::serialize::{SerializeAction, SerializeActionGenerator};
+use crate::deploy::actors::task_manager::{StopReason, TaskManagerInit, TaskSource};
+use crate::interlude::*;
+use clap::Parser;
+use common::typegraph::Typegraph;
+use core::fmt::Debug;
+
+use common::graphql::Query;
+use reqwest::header::{HeaderMap, HeaderValue, AUTHORIZATION};
+
+#[derive(Parser, Debug)]
+pub struct List {
+    #[clap(long)]
+    max_parallel_loads: Option<usize>,
+}
+
+#[async_trait]
+impl Action for List {
+    #[tracing::instrument]
+    async fn run(&self, args: ConfigArgs) -> Result<()> {
+        let dir = args.dir()?;
+        let config_path = args.config.clone();
+
+        let config = Config::load_or_find(config_path, &dir)?;
+
+        let config = Arc::new(config);
+
+        let console = ConsoleActor::new(Arc::clone(&config)).start();
+
+        let action_generator = SerializeActionGenerator::new(
+            config.dir().unwrap_or_log().into(),
+            dir.clone().into(),
+            config
+                .prisma_migrations_base_dir(PathOption::Absolute)
+                .into(),
+            true,
+        );
+
+        let task_source = TaskSource::Discovery(dir.into());
+
+        let mut init = TaskManagerInit::<SerializeAction>::new(
+            config.clone(),
+            action_generator,
+            console,
+            task_source,
+        );
+
+        if let Some(max_parallel_tasks) = self.max_parallel_loads {
+            init = init.max_parallel_tasks(max_parallel_tasks);
+        }
+
+        let report = init.run().await;
+
+        match report.stop_reason {
+            StopReason::Error => bail!("failed"),
+            StopReason::Manual | StopReason::ManualForced => {
+                bail!("cancelled")
+            }
+            StopReason::Natural => {}
+            StopReason::Restart => panic!("restart not supported for serialize"),
+        }
+
+        let tgs = report.into_typegraphs()?;
+
+        self.display_typegrahs(tgs).await?;
+        Ok(())
+    }
+}
+
+#[derive(Debug, Deserialize)]
+struct TypegraphObject {
+    name: String,
+    url: String,
+}
+
+impl List {
+    async fn get_more_info(&self) -> Result<Vec<TypegraphObject>> {
+        let client = reqwest::Client::new();
+
+        let mut headers = HeaderMap::new();
+        headers.insert(
+            AUTHORIZATION,
+            HeaderValue::from_str(&format!("Basic {}", "YWRtaW46cGFzc3dvcmQ="))?, // I don't know if it right, may be get from env
```

Comment:
thought: you can get this info from the config. You just `base64(`${user}:${password}`). Or better yet, use the `post` method on the `Node` type found in the `common` crate. Grep for other usages in the cli for examples.

Reply:
I already change it, but I wait @Natoandro to check why path in Typegraph is None


```rust
    async fn get_more_info(&self, dir: PathBuf) -> Result<Vec<TypegraphObject>> {
        let query = r#"
            query {
                typegraphs {
                    name
                    url
                }
            }
        "#;
        let mut node_config = NodeConfig::default();
        node_config.url = Url::parse("http://localhost:7891")?;
        let node = node_config.build(dir).await?;
        let response = node
            .post("/typegate")
            .unwrap()
            .gql(query.into(), None)
            .await?;

        response
            .data("typegraphs")
            .map_err(|err| anyhow::anyhow!(err))
    }

```